andela/mrm_api
# CON-73 Send notification to admins when a room is over booked

#### Description
This Pull request allows an administrator for the converge web app to receive notifications whenever a room has been overbooked.

#### Type of Change

#### How has this been tested


#### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/mrm_api.git`
2. Setup project according to Readme.md
3. checkout to branch `story/CON-73-overbooked-rooms-notification`
4. run the following query to create an event. Make sure the eventId and other details are valid:

#### NB
- make sure the calendarId is correct.
- make sure the numberOfParticipants is more than the room capacity.

```
mutation {
  eventCheckin(
    eventId: "er0vj3v",
    calendarId: "calender id",
    numberOfParticipants: 100,
    eventTitle: "First LFA On-Boarding",
    startTime: "2018-08-14T06:00:00-07:00",
    endTime: "2018-08-14T07:00:00-07:00"
  ){
    event {
      id
      eventTitle
      room {
        id
        name
      }
    }
  }
}
```

5. To test the sockets, create an HTML file `index.html` and fill it with the following HTML code and change the location in the channel string to whatever location the cancelled event is happening, ie from `notificatons-lagos` to `notifications-kampala`

```
<!DOCTYPE html>
  <html lang="en">
  <head>
    <title>Flask_Chat_App</title>
    <script
      src="https://code.jquery.com/jquery-3.4.1.min.js"
      integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo="
      crossorigin="anonymous"></script>
  </head>
  <body>

    <h3 id="#messages" style='color: #ccc;font-size: 30px;'>No message yet..</h3>
    <div class="message_holder"></div>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/1.7.3/socket.io.min.js"></script>
    <script type="text/javascript">
      $(document).ready(function() {
        var socket = io.connect('http://127.0.0.1:5000');
        socket.on('connect', function() {
          socket.send('User has connected!');
        });
        socket.on('notifications-lagos', function(msg) {
          console.log(msg)
          console.log('Received message');
        });
      });
    </script>
  </body>
  </html>
```

6. After creating this HTML file, open it in the browser and open the console,  fire the query to cancel another event and you'd see the event object in the console of the HTML page.

#### Checklist



#### JIRA
[CON-72](https://andela-apprenticeship.atlassian.net/jira/software/projects/CON/boards/15/backlog?label=backend&selectedIssue=CON-73)
